jctools-core/src/main/java/org/jctools/queues/MpmcUnboundedXaddArrayQueue.java

```diff
@@ -133,6 +132,16 @@ public E poll()
                     {
                         break;
                     }
+                } else if (ccChunkIndex < ciChunkIndex) {
```

Comment:
I still don't see the point of this optimization. If you miss read the chunk you are going to now load the pIndex too. If the queue is actually empty you'll get here soon enough anyhow. If the queue is near empty contended with many consumers then you'll see empty more often, but not worth the extra code IMO.

Reply:
The point is that you cannot answer back for emptiness until a full rotation happen, while you already have that information: more then optimization is a way to use an information we already have without awaiting more the necessary

Reply:
> If you miss read the chunk you are going to now load the pIndex too

But here you're not missing to read the right chunk, but the consumer in charge of rotating `consumerChunk` has won the `casConsumerIndex` (that should point to the second element in the new chunk), but is still awaiting *something* before rotating `consumerChunk` using `soConsumerChunk`.
In this case, the current code will continue to spin, while it shouldn't if the second element in the new chunk is not going to be set.

Reply:
I'm not sure if we're seeing the same issue/case here or I'm misreading (entirely possible) your comment...